sdk/src/main/java/com/mapbox/maps/renderer/egl/EGLConfigChooser.kt

```diff
@@ -88,28 +86,50 @@ internal class EGLConfigChooser constructor(
 
   private fun getNumberOfConfigurations(
     egl: EGL10,
-    display: EGLDisplay,
-    configAttributes: IntArray
+    display: EGLDisplay
   ): IntArray {
     val numConfigs = IntArray(1)
-    if (!egl.eglChooseConfig(display, configAttributes, null, 0, numConfigs)) {
-      Logger.e(
-        TAG,
-        String.format(
-          MAPBOX_LOCALE,
-          "eglChooseConfig returned error %d",
-          egl.eglGetError()
+    val initialSampleCount = antialiasingSampleCount
+    var suitableConfigsFound = false
+    while (!suitableConfigsFound) {
+      val success = egl.eglChooseConfig(display, configAttributes, null, 0, numConfigs)
+      if (!success || numConfigs[0] < 1) {
+        Logger.e(
+          TAG,
+          String.format(
+            MAPBOX_LOCALE,
+            "eglChooseConfig returned error %d",
+            egl.eglGetError()
+          )
         )
-      )
-      eglChooserSuccess = false
+        if (antialiasingSampleCount > 1) {
+          Logger.w(TAG, "Reducing sample count in 2 times for MSAA as EGL_SAMPLES=$antialiasingSampleCount is not supported")
+          antialiasingSampleCount /= 2
```

Comment:
what if the device's supported sample count is like 4, 8 etc, while the user defined 5? would this algorithm end up picking 0 in the end?

Reply:
@pengdev 5 / 2 = 2, 2 / 2 = 1 => config with MSAA disabled will be chosen. Dividing any number (> 1) by 2 will end up with 1.

Reply:
if user defined 5, while device supports 4 or 8 - it depends on the driver, from what I've seen it EGL will select sample count rounded up so in this case it will select 8.
We will write a warning in this case though, this logic is not changed in this PR